receiver/hostmetricsreceiver/factory.go

```diff
@@ -81,14 +96,33 @@ func createMetricsReceiver(
 	host.EnableBootTimeCache(true)
 	process.EnableBootTimeCache(true)
 
-	return scraperhelper.NewScraperControllerReceiver(
+	scraperHelper, err := scraperhelper.NewScraperControllerReceiver(
 		&oCfg.ControllerConfig,
 		set,
 		consumer,
 		addScraperOptions...,
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	r.Unwrap().(*hostMetricsReceiver).scraperHelper = scraperHelper
+	return r, nil
 }
 
+func createLogsReceiver(
+	_ context.Context, set receiver.Settings, cfg component.Config, consumer consumer.Logs,
+) (receiver.Logs, error) {
+	oCfg := cfg.(*Config)
+
+	r := receivers.GetOrAdd(
+		cfg, func() component.Component {
+			return newHostMetricsReceiver(oCfg, &set)
```

Comment:
We cannot have different factories and reuse the result between metrics and logs. They are completely different objects `hostMetricsReceiver` and one created with `scraperhelper.NewScraperControllerReceiver` helper

Reply:
I don't see a reason to use the shared receiver here